cmd/soroban-rpc/internal/test/simulate_transaction_test.go

```diff
@@ -306,6 +306,64 @@ func TestSimulateTransactionSucceeds(t *testing.T) {
 	assert.Equal(t, result, resultForRequestWithDifferentTxSource)
 }
 
+func TestSimulateTransactionWithAuth(t *testing.T) {
+	test := NewTest(t)
+
+	ch := jhttp.NewChannel(test.sorobanRPCURL(), nil)
+	client := jrpc2.NewClient(ch, nil)
+
+	sourceAccount := keypair.Root(StandaloneNetworkPassphrase)
+	address := sourceAccount.Address()
+	account := txnbuild.NewSimpleAccount(address, 0)
+
+	helloWorldContract := getHelloWorldContract(t)
+
+	params := preflightTransactionParams(t, client, txnbuild.TransactionParams{
+		SourceAccount:        &account,
+		IncrementSequenceNum: true,
+		Operations: []txnbuild.Operation{
+			createInstallContractCodeOperation(account.AccountID, helloWorldContract),
+		},
+		BaseFee: txnbuild.MinBaseFee,
+		Preconditions: txnbuild.Preconditions{
+			TimeBounds: txnbuild.NewInfiniteTimeout(),
+		},
+	})
+
+	tx, err := txnbuild.NewTransaction(params)
+	assert.NoError(t, err)
+	sendSuccessfulTransaction(t, client, sourceAccount, tx)
+
+	deployContractOp := createCreateContractOperation(t, address, helloWorldContract, StandaloneNetworkPassphrase)
+	deployContractParams := txnbuild.TransactionParams{
+		SourceAccount:        &account,
+		IncrementSequenceNum: true,
+		Operations: []txnbuild.Operation{
+			deployContractOp,
+		},
+		BaseFee: txnbuild.MinBaseFee,
+		Preconditions: txnbuild.Preconditions{
+			TimeBounds: txnbuild.NewInfiniteTimeout(),
+		},
+	}
+	response := simulateTransactionFromTxParams(t, client, deployContractParams)
+	require.NotEmpty(t, response.Results)
+	require.Len(t, response.Results[0].Auth, 1)
+	require.Empty(t, deployContractOp.Auth)
+
+	var auth xdr.SorobanAuthorizationEntry
+	assert.NoError(t, xdr.SafeUnmarshalBase64(response.Results[0].Auth[0], &auth))
+	require.Equal(t, auth.Credentials.Type, xdr.SorobanCredentialsTypeSorobanCredentialsSourceAccount)
+	deployContractOp.Auth = append(deployContractOp.Auth, auth)
+	deployContractParams.Operations = []txnbuild.Operation{deployContractOp}
+
+	// preflight deployContractOp with auth
+	deployContractParams = preflightTransactionParams(t, client, deployContractParams)
+	tx, err = txnbuild.NewTransaction(deployContractParams)
+	assert.NoError(t, err)
+	sendSuccessfulTransaction(t, client, sourceAccount, tx)
```

Comment:
was kind of expecting this to re-run simulation with the tx now having the auth present?

```
response := simulateTransactionFromTxParams(t, client, deployContractParams)
```

Reply:
why ? rerunning the simulateTransaction is only needed in the case of non-source account transaction ( i.e. where you need other address to sign it )

Reply:
ah ok, cool, yeah, this test is essentially replicating the same rpc call flow that cli does during `contract deploy` for coverage.